### PR TITLE
Add Python package apsg

### PR DESCRIPTION
**Pull request by ondrolexa**
APSG - python package for structural geologists

### CODE REVIEW

File: recipes/apsg/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 64f471819be605425e43608e094713c29d072b724683baa41bfd4e98af698688
+
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vv"
+
```
Comment (carlodri) [inline recipes/apsg/meta.yaml]: you can also add `entry_points` here as per https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#python-entry-points

Comment (carlodri) [reply recipes/apsg/meta.yaml]: I see you have https://github.com/ondrolexa/apsg/blob/master/setup.py#L33

Comment (ondrolexa) [reply recipes/apsg/meta.yaml]: Thanks, entry points used in setup.py added to recipe.

File: recipes/apsg/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - iapsg=apsg.shell:main
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
+    - numpy >=1.14
+    - scipy >=1.0
```
Comment (carlodri) [inline recipes/apsg/meta.yaml]: in `setup.py` these requirements are not version-constrained, do you need the constraints here? why?

**Comment by carlodri** [comment]
@ondrolexa happy to help! just one more comment and LGTM (but I don't have the wand of power ✨ to merge!). If you need to speed up things, ping conda-forge/staged-recipes .

File: recipes/apsg/meta.yaml (hunk 2 above)
Comment (ondrolexa) [reply recipes/apsg/meta.yaml]: I used these constraints originally on Travis for conda package build, but you are right it is likely redundant here. Thanks for the review.

**Comment by CJ-Wright** [comment]
@conda-forge/staged-recipes

File: recipes/apsg/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    - pip
+  run:
+    - python
+    - numpy
+    - scipy
+    - matplotlib
+
+test:
+  imports:
+    - apsg
```
Comment (CJ-Wright) [inline recipes/apsg/meta.yaml]: can you add command test which tests the CLI?

File: recipes/apsg/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: d2c0b8f68fa3dc68e44d6ee83e5b01d894b1f2567f89fc97ff00749528dcef06
+
+build:
+  number: 0
+  noarch: python
```
Comment (CJ-Wright) [inline recipes/apsg/meta.yaml]: I'm not certain if noarch supports `entry_points` currently, don't remove it just yet though. Someone wiser might know.

Comment (scopatz) [reply recipes/apsg/meta.yaml]: It does

Comment (ondrolexa) [reply recipes/apsg/meta.yaml]: It is not really CLI app. CLI is used just to open an interactive shell and import APSG. I'm not sure how to test it.

File: recipes/apsg/meta.yaml (hunk 3 above)
Comment (ondrolexa) [reply recipes/apsg/meta.yaml]: It is not really CLI app. CLI is used just to open an interactive shell and import APSG. I'm not sure how to test it.

Comment (CJ-Wright) [reply recipes/apsg/meta.yaml]: Is there a `--help` option?

Comment (ondrolexa) [reply recipes/apsg/meta.yaml]: No, there are no command line arguments implemented. The interactive shell is still experimental and could be improved in next versions, but now it is sort of undocumented feature.

**Comment by CJ-Wright** [comment]
LGTM thank you very much @ondrolexa

**Pull request merged**